plugins/source/aws/client/spec/tableoptions/table_options.go

```diff
@@ -16,8 +24,21 @@ type TableOptions struct {
 	ECSTasks               *ECSTaskAPIs            `json:"aws_ecs_cluster_tasks,omitempty"`
 }
 
-func (t TableOptions) Validate() error {
-	v := reflect.ValueOf(t)
+func (t *TableOptions) SetDefaults() {
+	v := reflect.ValueOf(*t)
+	for i := 0; i < v.NumField(); i++ {
+		table := v.Field(i).Interface()
+		if !reflect.ValueOf(table).IsNil() {
+			tableInput, ok := table.(defaultsSetter)
+			if ok {
+				tableInput.SetDefaults()
+			}
+		}
+	}
+}
```

Comment:
Why separate `SetDefaults` from `Validate`?

Reply:
Different semantics: one is for validation that the structure os OK to use, the other is to add the default values (presumably, prior to validation).
IMO, the validation shouldn't change the value (I'd even argue that it should accept value receiver, not a pointer one)